### ISSUE TITLE
setting incorrect volume

### ISSUE THREAD

**Issue by beeguy1234** [jishi/node-sonos-discovery#108]
In node-sonos-discovery/lib/models/Player.js (line 509), I propose changing

  if (/^[+\-]/.test(level)) {  level = this.state.volume + parseInt(level);  }

to 

  if (/^[+\-]/.test(level)) { level = parseInt(this.state.volume) + parseInt(level);  }

In my experience, state.volume gets handled as a string.  So volume 9 +1 becomes 91, giving me a heart attack.  
  
(Note : depending on how state.volume was set, it's either a string or an int.  So this bug occurs only in certain circumstances.  Also only when increasing volume (+), because node handles subtraction (-) using int type.)

**Comment by GermanBluefox** [comment]
Then better so 

```
if` (/^[+-]/.test(level)) { level = parseInt(this.state.volume, 10) + parseInt(level, 10); }
```

**Comment by neophob** [comment]
good catch @beeguy1234 - discovered that issue too but I suspected my app. your finding make sense and explains why it's so hard to find. when this happened the volume of my sonos was usually more than 80 or 90. and yes, I totally understand the heard attack thing, the adrenaline goes up pretty fast when volume is above 90 late at night ;)

**Comment by jishi** [comment]
Good catch. It is already being parseInt'd when evented from the player, but if you have previously set a volume from the http api it is set as string (and the event is ignored). So the correct place to patch this would be is rather here:

https://github.com/jishi/node-sonos-discovery/blob/master/lib/models/Player.js#L513

I'll try and fix it as soon as I get the opportunity.

**Comment by jishi** [comment]
@neophob I'll verify the group volume as well, but I think it's fine because it is recalculated and not stored raw.

**Comment by neophob** [comment]
maybe @beeguy1234 want's to do it's firs PullRequest?

**Comment by beeguy1234** [comment]
@neophob  Show me the manual though ;-)
That's right I know nothing.   Been coding for 1 user, which is myself...
Anyway thanks for the quick turnaround.  Hurray for open source!

**Comment by jishi** [comment]
Resolved in 1.6.1
